integration-tests/mockbean/src/test/java/io/quarkus/it/mockbean/GreetingResourceTest.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.it.mockbean;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.is;
+
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.MockBean;
+
+@QuarkusTest
+class GreetingResourceTest {
+
+    @MockBean
+    MessageService messageService;
+
+    @MockBean
+    SuffixService suffixService;
+
+    @Test
+    public void testGreet() {
+        Mockito.when(messageService.getMessage()).thenReturn("hi");
```

Comment:
So I wonder if we could just get rid of the injection part and provide an API like this:
```java
@Test
public void testGreet() {
   Mockito.when(MockBean.of(MessageService.class).getMessage()).thenReturn("hi");
   ...
}
```
... where `MockBean.of()` would basically do `QuarkusMock.installMockForType(Mockito.mock(MessageService.class), MessageService.class)` and return the mock instance (and possibly cache the instance or something like that).

Reply:
Hm, maybe it's not a good idea. The injection points can be reused and allow us to mark the beans unremovable. Another idea - if we do `@Inject @MockBean` the client proxies will be injected first and we could wrap it afterwards. `@MockBean` is ignored by ArC but allows `CreateMockitoMocksCallback` to wrap the stuff. The advantage of this approach would be (1) no need to mark the beans unremovable, (2) no need to hold the instance handles and destroy them correctly.

Reply:
Let me think about it and get back to you tomorrow :)

Reply:
Feel free to ping me to discuss the API ;-)

Reply:
If a bean is removable then then mocking it would have no effect anyway.

Reply:
@mkouba if we do `@Inject @MockBean` in the test, then how would the user control the mocks? 
My idea about `@MockBean` is that it should just be a Mockito mock that users already know how to handle without having to do anything else.  If we add `@Inject` in the mix, won't that complicate the mock usage on the user side?

Reply:
> If a bean is removable then then mocking it would have no effect anyway.

If a bean is removable then a null would be injected or an exception would be thrown - depends on the `CreateMockitoMocksCallback` impl. My point is that right now you can test a bean in a `@QuarkusTest` even if it would be normally removed in production because test class is considered a bean.

> if we do @Inject @MockBean in the test, then how would the user control the mocks?

Like I said - the client proxies could be injected by ArC and then wrapped in the `CreateMockitoMocksCallback` and the wrapped mocks could be set back to the fields. So you would do `Mockito.when(messageService.getMessage()).thenReturn("hi")` like you do now.

Reply:
@mkouba I agree completely about testing beans injected into a class - they are made unremovable because you are actually testing those beans.
In the mock case however, you aren't testing the injected mocks, you are testing whatever component uses the mocks - so making them unremovable should have no effect.
Am I missing something?

Reply:
You're probably right. It's a bit confusing though. We should probably add some docs and explain the intention...

Reply:
Yes of course, I haven't added any docs yet but I absolutely plan to do so.

Reply:
I added some documentation. WDYT @mkouba ?